base/sim/FairMCApplication.cxx

```diff
@@ -936,6 +922,30 @@ void FairMCApplication::InitGeometry()
         }
     }
     fModIter->Reset();
+}
+
+//_____________________________________________________________________________
+void FairMCApplication::InitGeometry()
+{
+    fState = FairMCApplicationState::kInitGeometry;
+
+    // ToBeDone
+    // Recently the InitGeometry is called twice from the G4VMC, This is a work around tell the problem get fixed in
+    // G4VMC
+    if (fGeometryIsInitialized)
+        return;
+
+    /// Initialize geometry
+
+    /** Register stack and detector collections*/
+    FairVolume* fv = 0;
+    Int_t id = 0;
+    fModIter->Reset();
+
+    // Register stack
+    if (fRootManager && fRootManager->GetInstanceId() == 0) {
```

Comment:
I assume, your check for the InstanceId is for checking that this is the main thread or something? What about using `fParent` for this?

```suggestion
    if (fRootManager && !fParent) {
```